Got null pointerException when running javafx:jlink

Hello running mvn javafx:jlink on my project I got the following NullpointerException
Caused by: java.lang.NullPointerException
    at org.openjfx.JavaFXBaseMojo.getExecutablePath (JavaFXBaseMojo.java:368)
    at org.openjfx.JavaFXJLinkMojo.execute (JavaFXJLinkMojo.java:154)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

Any hints on what is going from ?

Hello, 
here are my cases:

`
set JAVA_HOME="C:\PROGRA~1\AdoptOpenJDK\jdk-14.0.0.36-hotspot dummy"
mvn clean install
`
dummy""=="" was unexpected at this time.


`
set JAVA_HOME="C:\PROGRA~1\AdoptOpenJDK\jdk-14.0.0.36-hotspot"
mvn -version
`
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: C:\Users\denis\data\apache-maven-3.6.3\bin\..
Java version: 14, vendor: AdoptOpenJDK, runtime: C:\PROGRA~1\AdoptOpenJDK\jdk-14.0.0.36-hotspot
Default locale: en_GB, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

`
set JAVA_HOME="C:\PROGRA~1\AdoptOpenJDK\jdk-14.0.0.36-hotspot"
mvn clean install
`
[ERROR] Failed to execute goal org.openjfx:javafx-maven-plugin:0.0.4:jlink (default) on project fx-bo-common-blog: Execution default of goal org.openjfx:javafx-maven-plugin:0.0.4:jlink failed.: NullPointerException -> [Help 1]

`
set JAVA_HOME=C:\PROGRA~1\AdoptOpenJDK\jdk-14.0.0.36-hotspot
mvn clean install
`
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------

--- a/src/main/java/org/openjfx/JavaFXBaseMojo.java
+++ b/src/main/java/org/openjfx/JavaFXBaseMojo.java
@@ -464,7 +464,15 @@ abstract class JavaFXBaseMojo extends AbstractMojo {
     }
 
     private String getJavaHomeEnv(Map<String, String> enviro) {
-        return enviro.get("JAVA_HOME");
+        String javahome = enviro.get("JAVA_HOME");
+        if (javahome == null || javahome.isEmpty()) {
+            return null;
+        }
+
+        int pathStartIndex = javahome.charAt(0) == '"' ? 1 : 0;
+        int pathEndIndex = javahome.charAt(javahome.length() - 1) == '"' ? javahome.length() - 1 : javahome.length();
+
+        return javahome.substring(pathStartIndex, pathEndIndex);
     }
 
     private int executeCommandLine(Executor exec, final CommandLine commandLine, Map<String, String> enviro,

